src/frontend/components/UI/AuthModal/index.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useRef } from 'react'
+import styles from './index.module.scss'
+import { ModalAnimation } from '@hyperplay/ui'
+import { WebviewTag } from 'electron'
+import { observer } from 'mobx-react-lite'
+import authModalState from '../../../state/authModalState'
+import { ethers } from 'ethers'
+import extensionStore from '../../../store/ExtensionStore'
+import onboardingStore from '../../../store/OnboardingStore'
+
+// TODO: replace this with dev portal prod URL when it's ready
+const url =
+  'https://hyperplay-dev-git-feature-unified-auth-ui-valist.vercel.app/signin'
+
+const METAMASK_ALREADY_PROVIDED_ERROR_CODE = -32002
+
+const isTooManyRequestsError = (error: string) => {
+  return error.includes(METAMASK_ALREADY_PROVIDED_ERROR_CODE.toString())
+}
+
+const AuthModal = () => {
+  const webviewRef = useRef<WebviewTag>(null)
+
+  const handleAccountNotConnected = async () => {
+    const currentProvider = await window.api.getCurrentWeb3Provider()
+
+    if (!currentProvider) {
+      onboardingStore.openOnboarding()
+      return
+    }
+
+    // try to trigger metamask popup to connect account
+    try {
+      const provider = new ethers.BrowserProvider(window.ethereum)
+      await provider.getSigner()
+    } catch (e) {
```

Comment:
I think the flow you want here is 

if there is no connected provider or current provider, then call `onboardingStore.openOnboarding()` 

if there is no connected provider but there is a current provider, then call
```
const currentWeb3Provider = await window.api.getCurrentWeb3Provider()
onboardingStore.bootstrapOnboarding(context, currentWeb3Provider)
``` 

if there is a connected provider (there will always be a current provider too in this instance), then you're good and 
```
window.ethereum.request({ method: 'eth_requestAccounts' })
```

should return the accounts

Reply:
When clicking mm mobile or wallet connect, mm extension is still requesting to sign the SIWE message.

We should update the above flow to compare the connected/current provider to the requested provider from the auth modal